src/deployment/deploy.py

```diff
@@ -613,6 +613,11 @@ def assign_scaleset_identity_role(self) -> None:
         )
 
     def assign_user_access(self) -> None:
+        if self.results["client_id"]:
```

Comment:
I don't understand this condition
why skipping when the client_id is set ? Shouldn't we update that client_id instead ?

Reply:
<img width="730" alt="image" src="https://user-images.githubusercontent.com/74685766/144102739-f043bb41-8306-45dd-8642-ef3b635273c4.png">
This was the offending code. You can't call this function as an SP.

Reply:
Okay @chkeita, I actually just changed it to self.upgrade. The original logic did not work properly because client_id and client_secret are set regardless.